backstage/src/executor/map_site.ts

```diff
@@ -0,0 +1,132 @@
+import { GameContext } from '@app/executor/game_context'
+import {
+   MapBranch,
+   MapSite,
+   MapSiteCustomFuncSelector,
+   MapSiteIdentSelector,
+} from '@app/ruleset'
+import { randChoose, randPropValue } from '@app/util/rand'
+
+export class GeneratedSite {
+   site: MapSite
+   left?: GeneratedBranch
+   right?: GeneratedBranch
+
+   constructor(site: MapSite) {
+      this.site = site
+   }
+
+   isLeafSite() {
+      return this.left === undefined
+   }
+}
+
+export class GeneratedBranch {
+   desc: string
```

Comment:
use `MaybeTranslationKey` or `MaybeTranslatable` instead (the latter supports some "string template" semantics)

Reply:
after compilation?

Reply:
Sure.